Skins: transform qss icon urls into absolute paths

ronso0 opened the pull request:
work around https://bugs.launchpad.net/mixxx/+bug/1922966

actually we could use the `background` shorthand in qss instead of the affected `image` #3863 
but that is a lot of work with (see that PR description) and also doesn't work satisfactory for library widgets, since `background` wouldn't scale images to the actual button size (which varies depending on the library font and the width of the preview column for example).

This pr simply adds absolute file paths where we use **skin:**/colortheme/icons.

Holzhaus reviewed (approved):
LGTM, I can confirm that this doesn't break anything for me. But I'm also not affected by the bug that this is supposed to fix.

ywwg commented on src/skin/legacyskinparser.cpp:
```diff
@@ -2030,6 +2030,12 @@ void LegacySkinParser::setupWidget(const QDomNode& node,
         m_style.clear(); // only apply color scheme to the first widget
     }
     if (!style.isEmpty()) {
+        // This replaces relative icon urls in stylesheets (external qss or inline
+        // <Style> nodes) with absolute file paths.
+        // TODO Can be removed/disabled as soon as https://bugs.launchpad.net/mixxx/+bug/1922966
+        // is fixed on all target OS/distros, or rather all target OS have
+        // the fixed package in their repo.
```
https://bugs.kde.org/show_bug.cgi?id=434451

ywwg replied:
And any release of kiconthemes != 5.80 will work

ywwg replied:
(i.e., should we try to only apply this fix for cases where the package is installed and the wrong version?)

Holzhaus replied:
No, I don't think this breaks anything for unaffected distros. works fine on my machine.

ywwg reviewed (approved):
This works on Ubuntu 21.04.  Brilliant fix

ronso0 committed:
skin parser: transform relative into absolute icon paths...

to work around missing SVG icons when broken libKF5IconThemes5 5.80
is installed

ronso0 commented:
I mentioned the kde bug in the comment.

the replace function is now separate and used for the launch image as well as the actual skin stylesheet.
@ywwg please take another look.

> (i.e., should we try to only apply this fix for cases where the package is installed and the wrong version?)

thought about that, too, but I have nooo idea how to detect that.
any downside with this fix except that it creates a slightly bigger temporary QString?

btw there are other plugins possibly interfering --or possibly helpful-- like `qt5ct` (apply system styles to menubar) or some dbus plugin (prints `Debug [Main]: D-Bus global menu: no`)

Holzhaus commented on src/skin/legacyskinparser.cpp:
```diff
@@ -2262,3 +2266,15 @@ void LegacySkinParser::addShortcutToToolTip(WBaseWidget* pWidget,
     return m_pContext->selectString(node, "LaunchImageStyle");
 }
+
+QString LegacySkinParser::stylesheetAbsIconPaths(QString& style) {
```
Mutable references are confusing IMHO, but in this case it's okay I guess.

Holzhaus reviewed (commented):
Checks fail.

ronso0 commented:
> Checks fail.

ahhmm, yes...got distracted. one more Ctrl+S and it works.

ronso0 commented:
btw @ywwg the preferences icons were not affected? 'only' skin icons and the checkmarks in the manubar, right?

ywwg reviewed (approved):
still works after update

ywwg commented:
preferences look fine

ronso0 commented:
alright, thanks.

An affected user in the forum managed to build this PR on Ubuntu Studio 21.04 and noticed the issues I watched during my VM testing as well (Ubuntu 21.04): the launch image, logo and some icons are distorted.
https://mixxx.discourse.group/t/icons-not-showing-in-ubuntu-studio/22201/6

meeeh..
Let's merge this, better than no icons.
And I'll see if I can fix it. Maybe something with icon size not matching the button size.

Holzhaus reviewed (approved):
Thanks!

ywwg commented:
thanks so much for this!

ronso0 commented:
yeah, welcome!
Tango is broken though due to `skin:../Tango/` which is required for Tango64. 
have fix ready that I'll push tonight.
